lib/Graphs/ICFG.cpp

```diff
@@ -460,11 +472,22 @@ struct DOTGraphTraits<ICFG*> : public DOTGraphTraits<PAG*>
         if (IntraBlockNode* bNode = SVFUtil::dyn_cast<IntraBlockNode>(node))
         {
             rawstr << "IntraBlockNode ID: " << bNode->getId() << " \t";
-            PAG::PAGEdgeList&  edges = PAG::getPAG()->getInstPTAPAGEdgeList(bNode);
-            for (PAG::PAGEdgeList::iterator it = edges.begin(), eit = edges.end(); it != eit; ++it)
-            {
-                const PAGEdge* edge = *it;
-                rawstr << edge->toString();
+            if (Options::IncludePAGInICFGDump) {
+                PAG::PAGEdgeList&  edges = PAG::getPAG()->getInstPTAPAGEdgeList(bNode);
+                rawstr << value2String(bNode->getInst()) << " \t";
+                PAG::PAGEdgeList::iterator it = edges.begin(), eit = edges.end();
+                if (it == eit) {
+                    // No edges, so we need to dump the instruction.
+                    rawstr << value2String(bNode->getInst()) << " \t";
+                }  else {
+                    for (; it != eit; ++it)
+                    {
+                        const PAGEdge* edge = *it;
+                        rawstr << edge->toString();
```

Comment:
You have two `value2String` which may output duplicate instructions, making it unncessary hard for debugging.

Can you change it to something as below:

```
PAG::PAGEdgeList&  edges = PAG::getPAG()->getInstPTAPAGEdgeList(bNode);
if(edges.empty()){
       rawstr << value2String(bNode->getInst()) << " \t";
}
else {
       for (PAG::PAGEdgeList::iterator it = edges.begin(), eit = edges.end(); it != eit; ++it) {
                const PAGEdge* edge = *it;
                rawstr << edge->toString();
       }
}

```